components/jira/jira.app.mjs

```diff
@@ -1,11 +1,257 @@
+      // See Section 3.2 of
+      // https://developer.atlassian.com/cloud/jira/platform/oauth-2-authorization-code-grants-3lo-for-apps/
+      const resp = await axios($ ?? this, {
+        url: "https://api.atlassian.com/oauth/token/accessible-resources",
+        headers: {
+          Authorization: `Bearer ${this.$auth.oauth_access_token}`,
+        },
+      });
+      // Assumes the access token has access to a single instance
+      return resp[0].id;
+    },
+    _getHeaders(headers = {}) {
+      return {
+        "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "user-agent": "@PipedreamHQ/pipedream v0.1",
+        ...headers,
+      };
+    },
+    async _getUrl($, path) {
+      const cloudId = await this._getCloudId($);
+      return `https://api.atlassian.com/ex/jira/${cloudId}/rest/api/3${path}`;
+    },
+    async _makeRequest({
+      $,
+      path,
+      headers,
+      ...otherConfig
+    } = {}) {
+      const config = {
+        url: await this._getUrl($, path),
+        headers: this._getHeaders(headers),
+        ...otherConfig,
+      };
+      return axios($ ?? this, config);
+    },
+    async assignIssue({
+      issueIdOrKey,
+      ...args
+    } = {}) {
+      return await this._makeRequest({
+        method: "PUT",
+        path: `/issue/${issueIdOrKey}/assignee`,
+        ...args,
+      });
+    },
+    async addWatcher({
+      issueIdOrKey,
+      accountId,
+      ...args
+    } = {}) {
+      return await this._makeRequest({
+        method: "POST",
+        path: `/issue/${issueIdOrKey}/watchers`,
+        data: `"${accountId}"`,
+        ...args,
+      });
+    },
+    async addAttachmentToIssue({
+      issueIdOrKey,
+      ...args
+    } = {}) {
+      return await this._makeRequest({
+        method: "POST",
+        path: `/issue/${issueIdOrKey}/attachments`,
+        ...args,
+      });
+    },
+    async addCommentToIssue({
+      issueIdOrKey,
+      ...args
+    } = {}) {
+      return await this._makeRequest({
+        method: "POST",
+        path: `/issue/${issueIdOrKey}/comment`,
+        ...args,
+      });
+    },
+    async createIssue({ ...args } = {}) {
+      return await this._makeRequest({
+        method: "POST",
+        path: "/issue",
+        ...args,
+      });
+    },
+    async createVersion({ ...args } = {}) {
+      return await this._makeRequest({
+        method: "POST",
+        path: "/version",
+        ...args,
+      });
+    },
+    async deleteProject({
+      projectIdOrKey,
+      ...args
+    } = {}) {
+      return await this._makeRequest({
+        method: "DELETE",
+        path: `/project/${projectIdOrKey}`,
+        ...args,
+      });
+    },
+    async getAllProjects({ ...args } = {}) {
+      return await this._makeRequest({
+        method: "GET",
+        path: "/project",
+        ...args,
+      });
+    },
```

Comment:
This endpoint is actually deprecated. I think we should use `/project/search`
https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-projects/#api-rest-api-3-project-search-get
And make sure to handle pagination
https://developer.atlassian.com/cloud/jira/platform/rest/v3/intro/#pagination

Reply:
Sorry for that, I guess I missed it